sopel/modules/seen.py

```diff
@@ -63,9 +70,12 @@ def seen(bot, trigger):
 @plugin.require_chanmsg
 def note(bot, trigger):
     nick = trigger.nick
-    # as of Sopel 8, `trigger.time` is Aware, meaning we should store its value
-    # for timezone safety when comparing it later
-    bot.db.set_nick_value(nick, 'seen_timestamp', trigger.time.timestamp())
-    bot.db.set_nick_value(nick, 'seen_channel', trigger.sender)
-    bot.db.set_nick_value(nick, 'seen_message', trigger)
-    bot.db.set_nick_value(nick, 'seen_action', trigger.ctcp is not None)
+    try:
+        # as of Sopel 8, `trigger.time` is Aware, meaning we should store its value
+        # for timezone safety when comparing it later
+        bot.db.set_nick_value(nick, 'seen_timestamp', trigger.time.timestamp())
+        bot.db.set_nick_value(nick, 'seen_channel', trigger.sender)
+        bot.db.set_nick_value(nick, 'seen_message', trigger)
+        bot.db.set_nick_value(nick, 'seen_action', trigger.ctcp is not None)
+    except SQLAlchemyError:
+        logger.exception("Database write failed")
```

Comment:
At this point, I don't think the seen plugin should log the traceback, that would be very spammy without providing much of an information: if any of these method call fail, we should already know about it. The code is fairly stable, and at this point we all know it's about preventing log spam.

So I suggest to log the error, but not the traceback, with this:

```suggestion
    except SQLAlchemyError as error:
        logger.error("Unable to save seen, database error: %s" % error)
```